sdk/src/Core/Amazon.Runtime/Internal/Transform/JsonUnmarshallerContext.cs

```diff
@@ -231,74 +242,74 @@ public override bool Read()
-        /// </returns>
-        public override string ReadText()
+        public string ReadText(ref StreamingUtf8JsonReader reader)
         {
-            object data = jsonReader.Value;
-            string text;
+            string text = string.Empty;
             switch (currentToken)
             {
-                case JsonToken.Null:
+                case JsonTokenType.Null:
                     text = null;
                     break;
-                case JsonToken.String:
-                case JsonToken.PropertyName:
-                    text = data as string;
+                case JsonTokenType.String:
+                    text = reader.Reader.GetString();
+                    break;
+                case JsonTokenType.Number:
+                    if (reader.Reader.TryGetInt64(out long longValue))
+                        text = longValue.ToString(CultureInfo.InvariantCulture);
+                    else if (reader.Reader.TryGetDouble(out double doubleValue))
+                        text = doubleValue.ToString(CultureInfo.InvariantCulture);
+                    else if (reader.Reader.TryGetInt32(out int intValue))
+                        text = intValue.ToString(CultureInfo.InvariantCulture);
                     break;
-                case JsonToken.Boolean:
-                case JsonToken.Int:
-                case JsonToken.UInt:
-                case JsonToken.Long:
-                case JsonToken.ULong:
-                    IFormattable iformattable = data as IFormattable;
-                    if (iformattable != null)
-                        text = iformattable.ToString(null, CultureInfo.InvariantCulture);
-                    else
-                        text = data.ToString();
+                case JsonTokenType.PropertyName:
+                    text = reader.Reader.GetString();
                     break;
-                case JsonToken.Double:
-                    var formattable = data as IFormattable;
-                    if (formattable != null)
-                        text = formattable.ToString("R", CultureInfo.InvariantCulture);
-                    else
-                        text = data.ToString();
+                case JsonTokenType.True:
+                case JsonTokenType.False:
+                    text = reader.Reader.GetBoolean().ToString();
                     break;
                 default:
-                    throw new AmazonClientException(
-                            "We expected a VALUE token but got: " + currentToken);
+                    throw new AmazonClientException($"Unexpected token: {currentToken}");
             }
             return text;
         }
 
+        /// <summary>
+        ///     Returns the text contents of the current token being parsed.
+        /// </summary>
+        /// <returns>
+        ///     The text contents of the current token being parsed.
+        /// </returns>
+        public override string ReadText()
```

Comment:
See if you have to have this method defined on the base class. Any time we have to code like this check to see if the methods should really be on the base type.

Reply:
Oh good catch. This one I just forgot to remove. Since the method signatures are different, I removed it from `UnmarshallerContext`.